### ISSUE TITLE
BUG in computation of Hit@ X ?

### ISSUE THREAD

**Issue by unmeshvrije** [thunlp/OpenKE#235]
Looking at openke/base/Test.h and removing type_constrain related code for ease:
Please find comments in code that I wrote in order to understand it
```
extern "C"
void testHead(REAL *con, INT lastHead, bool type_constrain = false) {
    INT h = testList[lastHead].h;
    INT t = testList[lastHead].t;
    INT r = testList[lastHead].r;
    // This is a test triple (h, r, t)

    INT lef, rig;
    // minimal contains the *score* for h
    REAL minimal = con[h];

    INT l_s = 0;
    INT l_filter_s = 0;
    INT l_s_constrain = 0;
    INT l_filter_s_constrain = 0;

    for (INT j = 0; j < entityTotal; j++) {
        // For all entities that are not h
        if (j != h) {
            // find out the score of the considered entity
            REAL value = con[j];
            if (value < minimal) {
                // Why increment here ?
                // The entities are not even sorted.
                l_s += 1;

                if (not _find(j, t, r)){
                    // How come this works for the filtered settings ?
                    // _find() function in the Corrupt.h considers *ALL* triples
                    // AFAIK, filtered setting should not consider triples in the training/validation sets
                    l_filter_s += 1;
                } 
             // ....
     }
     
     // This cannot be considered a HIT, because l_s is incremented for any score that is < score(h)
     // _find() function is called after incrementing
     if (l_s < 10) l_tot += 1;
}
```

Am I mistaken in understanding the code. Please correct me in that case.

**Comment by THUCSTHanxu13** [comment]
hits@10 means that no more than 10 corrupted entities have smaller scores than the right entity. Hence, sorting entities is not necessary, and we directly compute the number of corrupted entities having smaller scores than the right entity. 

 _find(j, t, r) if to find if the triple (j,r,t) in the training set, test set, and validation set.
